evaluation/evaluate_agent.py

```diff
@@ -0,0 +1,245 @@
+                    else:
+                        episode_rewards.append(current_rewards[i])
+                        episode_lengths.append(current_lengths[i])
+                        episode_counts[i] += 1
+                    current_rewards[i] = 0
+                    current_lengths[i] = 0
+
+        observations = new_observations
+
+        if render:
+            frame = env.render()
+            episode_frames.append(frame)
+
+            if dones[0]:
+                all_episode_frames.append(episode_frames)
+                episode_frames = []
+
+    mean_reward = np.mean(episode_rewards)
+    std_reward = np.std(episode_rewards)
+    if reward_threshold is not None:
+        assert mean_reward > reward_threshold, (
+            "Mean reward below threshold: "
+            f"{mean_reward:.2f} < {reward_threshold:.2f}"
+        )
+    if return_episode_rewards:
+        return episode_rewards, episode_lengths, all_episode_frames
+    return mean_reward, std_reward
+
+
+def main(args):
+    checkpoints = args.checkpoint if args.checkpoint is not None else []
+    env_id = args.env_id
+    goal_changes = args.goal_changes
+    obj_to_hide = ["goal"] if args.hide_goals else []
+    stack_frames = args.stack_frames
+    num_eval_episodes = args.num_eval_episodes
+    seed = args.seed if args.seed is not None else np.random.randint(0, 1000000)
+    save = args.save
+    save_video = args.save_video
+
+    # eval env
+    eval_reward = reward_factory(reward="sparse")
+    eval_env = make_env(
+        env_id=env_id,
+        rank=0,
+        seed=seed,
+        reward_fn=eval_reward,
+        goal_changes=goal_changes,
+        obj_to_hide=obj_to_hide,
+        stack_frames=stack_frames,
+    )()
+
+    # video dir
+    if save:
+        videodir = f"evaluations-{datetime.now().strftime('%Y-%m-%d-%H-%M-%S')}"
+        os.makedirs(videodir, exist_ok=True)
+        with open(os.path.join(videodir, "args.txt"), "w") as f:
+            f.write(str(args) + "\n")
+
+    # load trained model
+    assert len(checkpoints) > 0 and all(
+        [c.endswith("zip") for c in checkpoints]
+    ), "invalid checkpoints"
+    for checkpoint in checkpoints:
+        model = PPO.load(checkpoint)
+
+        rewards, lengths, frames = custom_evaluate_policy(
+            model,
+            eval_env,
+            n_eval_episodes=num_eval_episodes,
+            render=save,
+            return_episode_rewards=True,
+        )
+        rewards = np.array(rewards)
+        lengths = np.array(lengths)
+        frames = [np.array(ep_frames) for ep_frames in frames]
+
+        print(
+            f"Model: {checkpoint}, eval env: {env_id}, num eval episodes: {num_eval_episodes}"
+        )
+        print(f"Mean reward: {rewards.mean():.2f} +/- {rewards.std():.2f}")
+        print(f"Mean length: {lengths.mean():.2f} +/- {lengths.std():.2f}")
+        print(f"Collected frames: {[f.shape for f in frames]}")
+        print()
+
+        # saving video
```

Comment:
This comment is incorrect as it prefaces the general save section. Would probably remove